api-client-util/src/main/java/com/takipi/api/client/util/cicd/ProcessQualityGates.java

```diff
@@ -0,0 +1,190 @@
+		}
+		
+		//filter out unneeded events set in plugin (regexFilter)
+		events = filterEvents(events, regexFilter);
+		
+		//check count gates
+		if (countGate) {
+			events = RateRegression.getSortedNewEvents(events);
+			qualityReport.setTopErrors(getTopXEvents(apiClient, input, events, topIssuesVolume, deploymentStart));
+		}
+		
+		//find new Errors
+		if (newEvents) {
+			qualityReport.setNewErrors(getNewErrors(apiClient, input, events, deploymentStart));
+		}
+		
+		//find resurfaced Errors
+		if (resurfacedEvents) {
+			qualityReport.setResurfacedErrors(getResurfacedErrors(apiClient, input, events, deploymentStart));
+		}
+		
+		//find critical errors
+		if (input.criticalExceptionTypes != null) {
+			qualityReport.setCriticalErrors(getCriticalErrors(apiClient, input, events, deploymentStart));
+		}
+		
+		return qualityReport;
+	}
+	
+	private static List<OOReportEvent> getTopXEvents(ApiClient apiClient, RegressionInput input,
+			Collection<EventResult> events, int topIssuesVolume, DateTime deploymentStart) {
+		List<OOReportEvent> result = new ArrayList<OOReportEvent>();
+		
+		int counter = 1;
+		for (EventResult event : events) {
+			String arcLink = getArcLink(apiClient, event.id, input, deploymentStart);
+			OOReportEvent newEvent = new OOReportEvent(event, null, arcLink);
+			result.add(newEvent);
+			
+			if (counter == topIssuesVolume) {
+				break;
+			}
+			counter++;
+		}
+		return result;
+	}
+	
+	//find critical errors from error list
+	private static List<OOReportEvent> getCriticalErrors(ApiClient apiClient, RegressionInput input,
+			Collection<EventResult> events, DateTime deploymentStart) {
+		List<OOReportEvent> result = new ArrayList<OOReportEvent>();
+		for (EventResult event : events) {
+			if (!input.criticalExceptionTypes.contains(event.name)) {
+				continue;
+			}
+			String arcLink = getArcLink(apiClient, event.id, input, deploymentStart);
+			OOReportEvent newEvent = new OOReportEvent(event, null, arcLink);
+			result.add(newEvent);
+		}
+		return result;
+	}
+	
+	//find resurfaced errors from error list
+	private static List<OOReportEvent> getResurfacedErrors(ApiClient apiClient, RegressionInput input,
+			Collection<EventResult> events, DateTime deploymentStart) {
+		List<OOReportEvent> result = new ArrayList<OOReportEvent>();
+		for (EventResult event : events) {
+			if (!event.labels.contains(RegressionStringUtil.RESURFACED_ISSUE)) {
+				continue;
+			}
+			String arcLink = getArcLink(apiClient, event.id, input, deploymentStart);
+			OOReportEvent newEvent = new OOReportEvent(event, null, arcLink);
+			result.add(newEvent);
+		}
+		return result;
+	}
+	
+	//find new errors from error list
+	private static List<OOReportEvent> getNewErrors (ApiClient apiClient, RegressionInput input, Collection<EventResult> events, DateTime deploymentStart) {		
+		List<OOReportEvent> result = new ArrayList<OOReportEvent>();		
+		for (EventResult event : events) {			
+			if (!input.deployments.contains(event.introduced_by)) {
```

Comment:
Probably better to use `!CollectionUtil.safeContains(input.deployments, event.introduced_by)`, as it protects us in case `input.deployments` is `null`

Reply:
There is a check before this code is called that makes sure there are events. No need to check again.